frontend/src/components/worksheets/NewUpload/NewUpload.jsx

```diff
@@ -41,54 +42,171 @@ class NewUpload extends React.Component<{
         percentComplete: 0,
     }
 
+    inputFolder = React.createRef();
     inputFile = React.createRef();
 
     setFile = () => {
         const files = this.inputFile.current.files;
+
+        if (!files.length) {
+            return;
+        }
+        this.uploadFile(files);
+    }
+
+    setFolder = () => {
+        const files = this.inputFolder.current.files;
         if (!files.length) {
             return;
         }
-        this.uploadFile(files[0]);
+        this.uploadFolder(files);
     }
 
-    uploadFile = (file) => {
-        if (!file) {
+    uploadFile = (files) => {
+        if (!files) {
+            return;
+        }
+        let folderSize = 0;

```

Comment:
You can instead do (looks nicer, also use const instead of let):

```
const folderSize = files.reduce((a, b) => a.size + b.size, 0);
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/FileList
`FileList API` is actually an object, not an array.